src/Psalm/Internal/Analyzer/StatementsAnalyzer.php

```diff
@@ -800,12 +816,201 @@ function ($line) {
     }
 
     /**
+     * @param  string   $var_id
+     * @param  CodeLocation   $var_loc
+     * @param  int  $end_bound
+     * @return FileManipulation
+     */
+    private function getPartialRemovalBounds(CodeLocation $var_loc, int $end_bound): FileManipulation
+    {
+        $var_start_loc= $var_loc->raw_file_start;
+        $stmt_content = $this->getSource()->getCodebase()->file_provider->getContents($var_loc->file_path);
+        $str_for_token = "<?php\n" . substr($stmt_content, $var_start_loc, $end_bound - $var_start_loc + 1);
+        $token_list = array_slice(token_get_all($str_for_token), 1);   //Ignore "<?php"
+
+
+        $offset_count = strlen($token_list[0][1]);
+
+        // Check if second token is just whitespace
+        if (is_array($token_list[1]) && strlen(trim($token_list[1][1])) == 0) {
+            $offset_count += strlen($token_list[1][1]);
+            $eq_char_index = 2;
+        } else {
+            $eq_char_index = 1;
+        }
+
+        // Add offset for assignment operator
+        if (is_string($token_list[$eq_char_index])) {
+            $offset_count += 1;
+        } else {
+            $offset_count += strlen($token_list[$eq_char_index][1]);
+        }
+
+        // Remove any whitespace following assignment operator token (e.g "=", "+=")
+        if (is_array($token_list[$eq_char_index + 1]) && strlen(trim($token_list[$eq_char_index + 1][1])) == 0) {
+            $offset_count += strlen($token_list[$eq_char_index + 1 ][1]);
+        }
+
+        $file_man_start = $var_start_loc;
+        $file_man_end = $var_start_loc + $offset_count;
+
+        return new FileManipulation($file_man_start, $file_man_end, "", false);
+    }
+
+    /**
+     * @param  PhpParser\Node\Expr\Assign|PhpParser\Node\Expr\AssignOp|PhpParser\Node\Expr\AssignRef $cur_assign
+     * @param  array<string, CodeLocation>    $var_loc_map
+     * @return void
+     */
+    private function markRemovedChainAssignVar(PhpParser\Node\Expr $cur_assign, array $var_loc_map): void
+    {
+        $var = $cur_assign->var;
+        if ($var instanceof PhpParser\Node\Expr\Variable && is_string($var->name)) {
+            $var_name = "$" . $var->name;
+            $var_loc = $var_loc_map[$var_name];
+            $this->removed_unref_vars[$var_name] = $var_loc;
+
+            $rhs_exp = $cur_assign->expr;
+            if ($rhs_exp instanceof PhpParser\Node\Expr\Assign ||
+                $rhs_exp instanceof PhpParser\Node\Expr\AssignOp ||
```

Comment:
small nit: mind putting `||` at the start of the line, vs the end of it? Doing so aids legibility a small amount.

So

```
            if ($rhs_exp instanceof PhpParser\Node\Expr\Assign
                || $rhs_exp instanceof PhpParser\Node\Expr\AssignOp
```

etc.